hooks/ctx.tsx

```diff
@@ -30,11 +31,20 @@ export function SessionProvider(props: React.PropsWithChildren) {
   return (
     <AuthContext.Provider
       value={{
-        signIn: () => {
-          // Add your login logic here
-          // For example purposes, we'll just set a fake session in storage
-          //This likely would be a JWT token or other session data
-          setSession("John Doe");
+        signIn: (email, password) => {
+          const data = {
+            user: {
+              email: email,
+              password: password
+            }
+          };
+          axios.post('https://rs-blackmarket-api.herokuapp.com/api/v1/users/sign_in', data)
+          .then(function (response) {
+            setSession(response.headers['authorization']);
+          })
+          .catch(function (error) {
```

Comment:
As an improvement of this, you can consider using react-query, several RN projects have been using it.
It simplifies a lot of handling API requests.

[Young Athletes](https://github.com/rootstrap/special-olympics-react-native)
[Fitness App](https://github.com/rootstrap/soifa-react-native)

Reply:
I will take a look into this thanks